### ISSUE TITLE
allow import/export of an instance's mod list as csv

### ISSUE THREAD

**Issue by microlith57** [PrismLauncher/PrismLauncher#214]
### Role

i play modded Minecraft

### Suggestion

it would be nice to be able to export an instance's mods (or a subset thereof) as a .csv of their name, version, etc., and to be able to import this by downloading the mods in bulk.

### Benefit

sometimes i want to share my modlist with others, but the only way to do this at the moment is by creating a modpack or by sending the .jar files; it would be much easier if a text-based mod list could be imported.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

currently Prism can install mods in bulk, but this requires first finding and selecting them in the 'download mods' window. the csv import would be an alternative way to perform this step.

**Comment by Scrumplex** [comment]
You could use something like @ryanccn's [Moddermore](https://moddermore.vercel.app/) to share your instance with others

**Comment by cfitz25** [comment]
id like this feature too, being able to export mod lists as a CSV or JSON file would make it easier to play with specific mods with friends as you could just send that file and prism could just use its normal mod downloading system to set it up

**Comment by sophuric** [comment]
also export to a simple text file, with just the mod names, for people who don't use prism launcher

**Comment by Trial97** [comment]
> also export to a simple text file, with just the mod names, for people who don't use prism launcher

The export part is already done in develop(no import, just the export to a modlist):
It supports:
- HTML
- Markdown
- Plain text
- JSON
- CSV

**Comment by TayouVR** [comment]
I don't think it makes sense to have import from text file like csv, as in that case people may just export to .mrpack or cf packs, which are essentially just formatted text files (depending on what you decide to include)

**Comment by ruguysgoingtrickortreating** [comment]
is it possible to do both though, modrinth + curseforge in one as of now?

**Comment by IdrisQe** [comment]
I think that any exportable format that has the required info to locate mods should also be importable, from the standpoint of intuitiveness.

Also yeah, as implied above, it would be nice to have the ability to export and import via a format that can use both modrinth and curseforge mods, but without needing overrides and without needing to use a 3rd-party utility like packwiz or moddermore.

**Comment by Trial97** [comment]
Nope as I have plans to move that out of export menu in order to allow exporting in a specified order and only the selected mods.

**Comment by IdrisQe** [comment]
So... no manifest-based modpack export/import that accepts both curseforge and modrinth links without overrides?

**Comment by Trial97** [comment]
Not as a modlist. 
Maybe there may be a chance of something similar in: https://github.com/PrismLauncher/PrismLauncher/issues/1182 where we want to do this for whole instance.

**Comment by devingDev** [comment]
I was disappointed that there is no import mod list feature since we installed mods from cf and modrinth...
It wouldn't be a problem if all mods were on modrinth but they aren't.

Although I think export as mrpack is alright since most mods we got are from modrinth and it just adds the cf jars(?)
not sure if this will work

Edit: Seems to work great although it doesnt pack the resourcepack/shaderpack i downloaded from modrinth in prism launcher (even if i tick the checkboxes of them in the export menu by myself)

**Comment by FeryFaithless** [comment]
idk why we cant import a list .txt or whtever for downloading easily or bring friend here from tl or..,why(!with names and versions and.. a simple list 4 download) its a hard thing for devs??or what's  reason.dorood 2 all ya specially devz and mod creatorz thnks alot,minecraft without mods is,,,, a,,,, weak thingy

**Comment by nheath** [comment]
I'd like to implement my own similar reporting, and would like to programatically get access to a list of profiles, installed mods and other profile settings.  In related issues i see lots of pushback to implement reporting in the launcher. One way to address this is t obe open for extension; allow other tools to generate these lists. Where is this underlying data stored? CF stores a json file with all the profile settings, mods, etc, and is ideal.

**Comment by Trial97** [comment]
Prism stores the mods  information in packwiz format inside `<path to instance>/mods/.index` 
The other information about the instance should be in `instance.cfg` and `mmc-pack.json`

**Comment by TRr4M** [comment]
bump. im not sure about specifically csv, but having an easy way to give your modpack to a friend without having to transfer a full zip which is too large for messaging platforms

**Comment by Trial97** [comment]
Closing for now as this is not currently planned or feasible.
However, this does not mean the idea is permanently rejected.
If any maintainer believes this is still feasible or worth pursuing, feel free to reopen the issue for further discussion.
